iis-wf/iis-wf-referenceextraction/src/main/java/eu/dnetlib/iis/wf/referenceextraction/softwareurl/CachedWebCrawlerJob.java

```diff
@@ -62,156 +59,88 @@ public static void main(String[] args) throws Exception {
         JCommander jcommander = new JCommander(params);
         jcommander.parse(args);
         
-        final String cacheRootDir = normalizePath(params.cacheRootDir);
-        
         ContentRetrieverContext contentRetrieverContext = new ContentRetrieverContext(params.contentRetrieverClassName, 
                 params.connectionTimeout, params.readTimeout, params.maxPageContentLength, params.numberOfEmittedFiles);
         
-        CacheMetadataManagingProcess cacheManager = new CacheMetadataManagingProcess();
-        
         try (JavaSparkContext sc = JavaSparkContextFactory.withConfAndKryo(new SparkConf())) {
-            HdfsUtils.remove(sc.hadoopConfiguration(), params.outputPath);
-            HdfsUtils.remove(sc.hadoopConfiguration(), params.outputFaultPath);
-            HdfsUtils.remove(sc.hadoopConfiguration(), params.outputReportPath);
-
-            LockManager lockManager = LockManagerUtils.instantiateLockManager(params.lockManagerFactoryClassName,
-                    sc.hadoopConfiguration());
             
-            OutputPaths outputPaths = new OutputPaths(params);
+            Configuration hadoopConf = sc.hadoopConfiguration();
+            
+            HdfsUtils.remove(hadoopConf, params.getOutputPath());
+            HdfsUtils.remove(hadoopConf, params.getOutputFaultPath());
+            HdfsUtils.remove(hadoopConf, params.getOutputReportPath());
+
+            LockManager lockManager = LockManagerUtils.instantiateLockManager(params.getLockManagerFactoryClassName(),
+                    hadoopConf);
             
             JavaRDD<DocumentToSoftwareUrl> documentToSoftwareUrl = avroLoader.loadJavaRDD(sc, params.inputPath,
                     DocumentToSoftwareUrl.class);
             
-            if (documentToSoftwareUrl.isEmpty()) {
-                storeInOutput(sc.emptyRDD(), sc.emptyRDD(),
-                        generateReportEntries(sc, sc.emptyRDD(), sc.emptyRDD(), sc.emptyRDD()), 
-                        outputPaths, contentRetrieverContext.getNumberOfEmittedFiles());
-                return;
-            }
+            final String cacheRootDir = CacheStorageUtils.normalizePath(params.getCacheRootDir());
+            CacheMetadataManagingProcess cacheManager = new CacheMetadataManagingProcess();
+            String existingCacheId = cacheManager.getExistingCacheId(hadoopConf, cacheRootDir);
             
-            // getting existing cache id
-            String existingCacheId = cacheManager.getExistingCacheId(sc.hadoopConfiguration(), cacheRootDir);
+            // skipping already extracted
+            JavaRDD<DocumentText> cachedSources = CacheStorageUtils.getRddOrEmpty(sc, avroLoader, cacheRootDir,
+                    existingCacheId, CacheRecordType.text, DocumentText.class);
+            // will be written in new cache version and output
+            cachedSources.cache();
```

Comment:
Yes, the comment is quite valid, I have just checked and we have quite a lot of contents stored in webcrawl cache: ~6GB.

Instead of changing it to `persist(StorageLevel.DISK_ONLY())` I would simply drop it though. There is no point in storing the whole RDD on a disk knowing it was read from the disk just one step before.